.github/workflows/build-plugin-zip.yml

```diff
@@ -201,6 +209,51 @@ jobs:
                   name: release-notes
                   path: ./release-notes.txt
 
+    revert-version-bump:
+        name: Revert version bump if build failed
+        needs: [bump-version, build]
+        runs-on: ubuntu-latest
+        if: |
+            always() &&
+            ( needs.build.outputs.job_status == 'failure' ) &&
+            needs.bump-version.outputs.release_branch_commit
+
+        steps:
+            - name: Checkout code
+              uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f # v2.3.4
+              with:
+                  fetch-depth: 2
+                  ref: ${{ needs.bump-version.outputs.release_branch }}
+                  token: ${{ secrets.GUTENBERG_TOKEN }}
+
+            - name: Configure git user name and email
+              run: |
+                  git config user.name "Gutenberg Repository Automation"
+                  git config user.email gutenberg@wordpress.org
+
+            - name: Revert version bump commit on release branch
+              if: |
+                  github.event.inputs.version == 'stable' ||
+                  contains( needs.bump-version.outputs.old_version, 'rc' )
+              run: |
+                  git revert --no-edit ${{ needs.bump-version.outputs.release_branch_commit }}
+                  git push --set-upstream origin "${{ needs.bump-version.outputs.release_branch }}"
+
+            - name: Delete release branch if it was only just created for the RC
+              if: |
+                  github.event.inputs.version == 'rc' &&
```

Comment:
Would this ever cause problems after the first RC? For example, a failure is encountered when bumping to RC2, 3, 4 etc.